test/parallel/test-fs-long-path.js

```diff
@@ -1,34 +1,46 @@
 'use strict';
-var common = require('../common');
-var fs = require('fs');
-var path = require('path');
-var assert = require('assert');
+const common = require('../common');
+const fs = require('fs');
+const path = require('path');
+const os = require('os');
 
-var successes = 0;
+// when it fails test again under real OS tmp dir on Linux when it is
+// readable/writable to avoid failing tests on ecryptfs filesystems:
+// https://github.com/nodejs/node/issues/2255
+// it follows advice in comments to:
+// https://github.com/nodejs/node/pull/3925
+// https://github.com/nodejs/node/pull/3929
+try {
+  common.refreshTmpDir();
+  testFsLongPath(common.tmpDir);
+  common.refreshTmpDir();
+} catch (e) {
+  if (os.type() == 'Linux') {
```

Comment:
`process.platform` is a little more idiomatic, at least w.r.t. to the test suite.